numpy/core/src/private/ufunc_override.c

```diff
@@ -170,6 +171,47 @@ normalize_at_args(PyUFuncObject *ufunc, PyObject *args,
 }
 
 /*
+ * Check whether an object has __array_ufunc__ defined on its class and it
+ * is not the default, i.e., the object is not an ndarray, and its
+ * __array_ufunc__ is not the same as that of ndarray.
+ *
+ * Note that since this module is used with both multiarray and umath, we do
+ * not have access to PyArray_Type and therewith neither to PyArray_CheckExact
+ * nor to the default __array_ufunc__ method, so instead we import locally.
+ * TODO: Can this really not be done more smartly?
+ */
+static int
+has_non_default_array_ufunc(PyObject *obj)
```

Comment:
It checks two things: whether the class has `__array_ufunc__` and, if so, whether it is different from that of `ndarray`...